numba/typed/typeddict.py

```diff
@@ -117,6 +117,33 @@ def __init__(self, **kwargs):
         else:
             self._dict_type = None
 
+        if args:
+            # CPython checks for at most 1 argument
+            # https://github.com/python/cpython/blob/f215d7cac9a6f9b51ba864e4252686dee4e45d64/Objects/dictobject.c#L2693-L2695
+            _len = len(args)
+            if _len > 1:
+                raise TypeError(f"Dict expect at most 1 argument, got {_len}")
+
+            # check if argument is iterable
+            arg = args[0]
+            if not isinstance(arg, Iterable):
+                msg = (f"'{type(arg)}' object is not iterable. Supported type "
+                       "constructor is Dict(iterable)")
+                raise TypeError(msg)
+
+            if isinstance(arg, Mapping):
+                for k, v in arg.items():
+                    self.__setitem__(k, v)
```

Comment:
This behaviour isn't working under JIT?

Reply:
I'll leave `dict(mapping)` ctor for a future PR and focus exclusively on `dict(iterable)` in this one.